src/modules/auth/infrastructure/cognito/aws.cognito.service.ts

```diff
@@ -30,19 +36,55 @@ export class CognitoService implements ICognitoService {
       endpoint: configService.get('AWS_COGNITO_ENDPOINT'),
     });
   }
-  registerAccount(email: string, password: string): Promise<IRegisterResult> {
-    return new Promise((resolve, reject) => {
-      this.userPool.signUp(email, password, null, null, (error, result) => {
-        if (!result) {
-          reject(error);
-        } else {
-          resolve({
-            externalId: result.userSub,
-            username: result.user.getUsername(),
+  async registerAccount(
+    email: string,
+    password: string,
+  ): Promise<IRegisterResult> {
+    try {
+      const signUpResult: ISignUpResult = await new Promise(
+        (resolve, reject) => {
+          this.userPool.signUp(email, password, null, null, (err, result) => {
+            if (!result) {
+              reject(err);
+            } else {
+              resolve(result);
+            }
           });
-        }
-      });
-    });
+        },
+      );
+
+      if (process.env.NODE_ENV === ENVIRONMENT.DEVELOPMENT) {
+        this.confirmAccount(email, signUpResult);
+      }
```

Comment:
We should check here that the env is not prod, in case we have different envs in the future